crates/control/src/kick_target_provider.rs

```diff
@@ -95,33 +118,82 @@ fn collect_kick_targets(
     field_dimensions: &FieldDimensions,
     obstacle_circles: &[Circle<Ground>],
     ball_position: Point2<Ground>,
-    max_kick_around_obstacle_angle: f32,
-    parameters: &FindKickTargetsParameters,
-    corner_kick_strength: f32,
-) -> Vec<KickTarget> {
+    collect_kick_targets_parameters: CollectKickTargetsParameter<'_>,
+    filtered_game_controller_state: Option<&FilteredGameControllerState>,
+) -> (Vec<KickTargetWithKickVariants>, bool) {
     let field_to_ground = ground_to_field.inverse();
 
-    let kick_targets = if is_ball_in_opponents_corners(
+    let is_own_kick_off = matches!(
+        filtered_game_controller_state.map(|x| x.game_state),
+        Some(FilteredGameState::Playing { kick_off: true, .. })
+    );
+    let is_not_free_for_opponent = matches!(
+        filtered_game_controller_state.map(|x| x.opponent_game_state),
+        Some(FilteredGameState::Playing {
+            ball_is_free: false,
+            ..
+        })
+    );
+
+    let (kick_opportunities, allow_instant_kicks): (Vec<_>, _) = if is_own_kick_off
+        && is_not_free_for_opponent
+    {
+        (
+            generate_kick_off_kick_targets(
+                field_dimensions,
+                field_to_ground,
+                collect_kick_targets_parameters.kick_off_kick_strength,
+            )
+            .into_iter()
+            .map(|kick_target| KickTargetWithKickVariants {
+                kick_target,
+                kick_variants: vec![KickVariant::Forward, KickVariant::Side],
```

Comment:
please make this a parameter. Why don't we want turn kicks here?

Reply:
I thought turn kicks are not as precise and as we have enough time at the kick off I think it's fine to joggle a bit but have a kick inside the right region.